sale_category_discount/models/sale_order_line.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import models, fields, api
+import odoo.addons.decimal_precision as dp
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    price_categ_id = fields.Many2one(
+        comodel_name='product.category',
+        compute='_get_price_categ_id',
+        store=True,
+        string='Category for Pricing',
+    )
+    price_categ_qty = fields.Float(
+        compute='_compute_price_categ_qty',
+        store=True,
+        string='Category Qty for Pricing',
+    )
+    price_unit = fields.Float(
+        'Unit Price',
+        required=True,
+        digits=dp.get_precision('Product Price'),
+        default=0.0,
+        compute='_recompute_price_unit',
+        store=True,
+    )
+
+    @api.multi
+    @api.depends('product_id')
+    def _get_price_categ_id(self):
+        for l in self.filtered('product_id'):
+            categs = l.order_id.pricelist_id.item_ids.filtered(
+                lambda x: x.applied_on == '2_product_category' and
```

Comment:
I'm concerned about the case user switches `applied_on` from `2_product_category` to something else, and `categ_id` field remains uncleared (haven't checked the behavior).